### ISSUE TITLE
missing feature for golang timers

### ISSUE THREAD

**Issue by zeroecco** [nanovms/nanos#1787]
The error coming from our app, this is leading to what we think is an ever growing number of go routines that never close.

```timer_create error: timer_create: clockid 0 not implemented```

We tried the the latest nightly and also attempted rebuilding the app a number of times.
The stack trace is as follows:

```timer_create error: timer_create: clockid 0 not implemented
timer_create error: timer_create: clockid 0 not implemented
timer_create error: timer_create: clockid 0 not implemented
timer_create error: timer_create: clockid 0 not implemented

frame trace: 
ffffc00000a37e20:   ffffffff800bdd54    (dispatch_signals + 0000000000000034/000000000000035c)
ffffc00000a37f10:   ffffffff800d9fbc    (thread_return + 000000000000012c/0000000000000543)
ffffc00000a37f40:   ffffffff80052df9    (runloop_internal + 0000000000000509/000000000000085e)
ffffc00000a37fc0:   ffffffff80041c01    (context_switch_finish + 0000000000000081/00000000000001c3)
assertion l failed at /home/circleci/project/src/unix/signal.c:117  in sigstate_dequeue_signal(); halt

frame trace: 
ffffc00000a37cb0:   ffffffff800b46a4    (notify_dispatch_with_arg + 0000000000000084/0000000000000141)
ffffc00000a37d00:   ffffffff8005c5d3    (tcp_input_lower + 00000000000000f3/000000000000016a)
ffffc00000a37d20:   ffffffff8012933c    (tcp_input + 0000000000000c0c/000000000000166c)
ffffc00000a37de0:   ffffffff8011789b    (ip4_input + 000000000000039b/00000000000003b0)
ffffc00000a37e30:   ffffffff80130176    (ethernet_input + 0000000000000126/0000000000000185)
ffffc00000a37e60:   ffffffff8002f607    (ena_cleanup + 0000000000000867/0000000000000b17)
ffffc00000a37f40:   ffffffff80052a73    (runloop_internal + 0000000000000183/000000000000085e)
ffffc00000a37fc0:   ffffffff80041c01    (context_switch_finish + 0000000000000081/00000000000001c3)
assertion !ctx->fault_handler failed at /home/circleci/project/src/kernel/kernel.h:564  in use_fault_handler(); halt```

**Comment by sanderssj** [comment]
How many vcpus does your guest have when running this program?

**Comment by zeroecco** [comment]
4 vCPU's (c6a.xlarge)

**Comment by francescolavra** [comment]
The first assertion failure in the above trace has been fixed in #1789.
The second assertion failure is a consequence of the first one (the fault handler of the context is non-zero because it has been set to a thread fault handler by the thread_return() function just before the first assertion failure occurs).

The timer_create errors are fixed in #1790, where support for additional clock types has been added in the Unix timer code. If you want to test these new features wthout waiting for this PR to be merged, you can insert the `--nanos-version f242845` option in the Ops command line to create an image.

**Comment by zeroecco** [comment]
This is amazing, thank you so much!
